src/Nncase.Core/Utilities/ReplaceUtility.cs

```diff
@@ -115,6 +115,61 @@ public static Call ReplaceParams(Call call, params (ParameterInfo, Expr)[] posAn
         return call with { Parameters = ReplaceMulti(call.Parameters, posAndValue) };
     }
 
+    /// <summary>
+    /// find the old input in old args and replace it with new_input.
+    /// </summary>
+    /// <param name="oldParams">matched old args.</param>
+    /// <param name="target">matched old input.</param>
+    /// <param name="newParam">created new_input.</param>
+    /// <returns>new args list.</returns>
+    /// <exception cref="InvalidOperationException"></exception>
+    public static List<Expr> ReplaceParams(IReadOnlyList<Expr> oldParams, Expr target, Expr newParam)
+    {
+        return ReplaceParams(oldParams, new List<Expr>() { target }, new List<Expr>() { newParam });
+    }
+
+    /// <summary>
+    ///  find the old input in old args and replace it with new_input.
+    /// </summary>
+    /// <param name="oldParams">matched old args.</param>
+    /// <param name="targets">matched old inputs.</param>
+    /// <param name="newParams">created new_inputs.</param>
+    /// <returns>new args list.</returns>
+    /// <exception cref="InvalidOperationException"></exception>
+    public static List<Expr> ReplaceParams(IReadOnlyList<Expr> oldParams, IReadOnlyList<Expr> targets, IReadOnlyList<Expr> newParams)
```

Comment:
It is recommended to use IReadOnlyList<target, newParam> or IDictionary<target, newParam>, the length of targets and newParams need to be the same and  have a correspondence